### PR TITLE
Fix Cars in doors

### LINKED ISSUE #6725: Doors have graphical glitches when vehicles go through
**OS:** Windows 10 / ArchLinux
**Version:** 0.1.2
**Commit/Build:** aff7609b86bc333a044711856b2510d7a3f07b97

Scenery wall doors have graphical glitches when vehicles go through them. Depending on the direction they are placed relative to the vehicle, the doors will open outwards then close inwards (as if it was placed the other way). The vehicle also gets drawn over the door with visible z-fighting.

This issue pre-dates v0.1.0.

### PR DESCRIPTION
**Pull request by duncanspumpkin**
image

~~Should probably test this a bit as I'm unsure why CS wrote it as a parent / child doesn't make any sense to me.~~

~~#6725 suggests this worked in vanilla but not sure how.~~

Edit: AHHH i see what went wrong weve used the wrong flag now matches vanilla.

Fix #6725

### CODE REVIEW

File: src/openrct2/paint/tile_element/Paint.Wall.cpp [hunk 1]
```diff
@@ -89,46 +81,41 @@ static void PaintWallDoor(
         case 0:
         {
             BoundBoxXYZ bbR1 = { { 1, 1, height + 1 }, { 1, 3, bbHeight - 5 } };
-            BoundBoxXYZ bbR2 = { { 1, 1, height + bbHeight - 9 }, { 1, 28, 3 } };
-
-            BoundBoxXYZ bbL = { { 1, 1, height + 1 }, { 1, 28, bbHeight } };
+            BoundBoxXYZ bbR2 = { { 1, 1, height + bbHeight - 4 }, { 1, 28, 3 } };
```
Comment (duncanspumpkin) [inline src/openrct2/paint/tile_element/Paint.Wall.cpp:84]: these offsets were wrong when we implemented the function. I checked it against the assembly implementor missed that there was a pop eax.

**Review by ZehMatt — APPROVED** [review]

**Comment by leicestersquare** [comment]
Thanks for fixing this!
While testing i found this not beeing resolved while looking in the directions of north and east when viewing the door from behind:
grafik
grafik

While working perfectly when looking in the direction of west and south.
grafik
grafik

Might not be realated though.

**Comment by duncanspumpkin** [comment]
> Thanks for fixing this! While testing i found this not beeing resolved while looking in the directions of north and east when viewing the door from behind: grafik grafik
> 
> While working perfectly when looking in the direction of west and south. grafik grafik
> 
> Might not be realated though.

Could you check that in vanilla rct2. I'm pretty sure that is just a standard z fighting issue unfortunately rather than what this is fixing.

**Comment by leicestersquare** [comment]
Ah great!
Yes. There is a similar bug in original rct2 vanilla.
From what i am guessing a Z-fighting issue would look like this then:
Door_Z_issue

**Comment by leicestersquare** [comment]
Also this situation is possible:
grafik

**Comment by Gymnasiast** [comment]
@leicestersquare Just to be sure: this PR does not make anything worse than it previously was, right? I’d like to merge it as it seems to make things better.

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**